### PR TITLE
Added noncentral_f random function to numpy frontend

### LINKED ISSUE #21602: noncentral_f

### PR DESCRIPTION
**Pull request by Honeson**
- [ ] Close #21602

- [ ] Opened an issue from the available issues for my contribution. The issue is on numpy frontend
- [ ] I created a random sampling function called **noncentral_f** for numpy frontend in ivy/functional/frontend/numpy/random/functions.py
- [ ] Also created a test for it called test_numpy_noncentral_f in ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py
- [ ] Kindly review to merge or let me know what I can change, so I can proceed to contributing to more issues.
- [ ] Thank you.

### CODE REVIEW

**Comment by Honeson** [comment]
Hello @Mr-Niraj-Kulkarni ,

Can you please review this my PR

Thanks

**Comment by Mr-Niraj-Kulkarni** [comment]
> Hello @Mr-Niraj-Kulkarni ,
> 
> Can you please review this my PR
> 
> Thanks

Hello, Sorry for late reply. I checked your PR have added some comments, Also the tests for implemented function are failing can you please look into that. Thanks!

**Comment by Honeson** [comment]
Hi @Mr-Niraj-Kulkarni 
I have corrected the issues and the tests are now passing 100%. 
Kindly review, thank you.

**Comment by Mr-Niraj-Kulkarni** [comment]
@Honeson  Hey can you please resolve the lint errors. Once Done I will merge the everything else looks good thanks

**Comment by Honeson** [comment]
> 

@Mr-Niraj-Kulkarni ,
I have resolved the linting errors now.

**Comment by Mr-Niraj-Kulkarni** [comment]
> > 
> 
> @Mr-Niraj-Kulkarni , I have resolved the linting errors now.

@Honeson  THere are unnecessary file into your PR please remove those. Thanks

**Comment by Honeson** [comment]
> 

@Mr-Niraj-Kulkarni ,
I have removed unnecessary files and pushed. Kindly review and if you find any irrelevant files again, please let me know the files so I can also remove them. Thanks.

**Comment by Mr-Niraj-Kulkarni** [comment]
@Honeson  I see still some file that shouldn't be there. 
while adding the file please don't do` git add.` Instead check for relevant files for your PR and  add only them by `git add file1 file2`
I hope this is helpful for you.

**Comment by Honeson** [comment]
> @Honeson I see still some file that shouldn't be there. while adding the file please don't do` git add.` Instead check for relevant files for your PR and add only them by `git add file1 file2` I hope this is helpful for you.

@Mr-Niraj-Kulkarni ,
Yes, that helps. I have staged and committed only the relevant files to my PR. Please check. Thank you.

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,
There seems to be changes made in the array api submodule, can you remove those changes.
And also in the .gitignore file

Thanks

**Comment by Honeson** [comment]
> Hi, There seems to be changes made in the array api submodule, can you remove those changes. And also in the .gitignore file
> 
> Thanks

Hi @sherry30 ,

I have addressed you issues you raised by adding only relevant files to my commit. Kindly check, thank you.

**Review by vedpatwardhan — COMMENTED** [review]
Just removed the auto-requested reviews 👍

File: ivy/functional/frontends/numpy/random/functions.py [hunk 1]
```diff
@@ -310,8 +334,6 @@ def triangular(left, mode, right, size=None):
     return ivy.where(condition, values1, values2)
 
 
-@to_ivy_arrays_and_back
```
Comment (vedpatwardhan) [inline ivy/functional/frontends/numpy/random/functions.py:313]: could you please revert this change and also the changes to the array api submodule? Thanks @Honeson 😄
(would be great if you could take it forward with the PR @NripeshN, thanks!)

**Review by vedpatwardhan — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/numpy/random/functions.py (hunk 1 above)
Comment (Honeson) [reply ivy/functional/frontends/numpy/random/functions.py:313]: Hi @vedpatwardhan 

I have made the changes you requested and created a new PR to keep things cleaner. Here is a link to the new PR for your review: https://github.com/unifyai/ivy/pull/27140

I will appreciate your feedback also @NripeshN.

Thank you.
